README.md

```diff
@@ -1,101 +1,95 @@
-On generic" boards, as this is always PA12 - as USB D+ is always on this pin on STM32F103 devices.
-(Note. For generic boards, this define could probably be moved to the usb code as it never changes)
-
-Note.
-USER_CODE_RAM and RAM_END are allways the same, and probably need to be consolodated, rather than being defined for each board.
-
-There is no longer a need to configure FLASH SIZE, as this is now detemined by reading the size from a register in the processor.
-The Flash page (and hence DFU upload block size) no longer needs to be defined either, as the flash page size is inferred from the total flash size. All devices up to 128kb have a 1k page size, and any board with more then 128k flash have a 2k page size.
-
-Note the code directly links flash page size to DFU block size. It would probably be possible to decouple this and possibly increase upload speeds, but it has not been done as yet.
-
+Let's go over the above.
 
-#####Other improvements on the original Maple bootloader
+1. Boards which have the Maple USB reset hardware need to define `HAS_MAPLE_HARDWARE`.
 
-1. Smaller footprint - now only 8k (This became possible due to code changes by @jonatanolofsson which allowed the GCC optimisation for size flag to be used
-2. Additional DFU AltID upload type was added, which allows the sketch to be loaded at 0x8002000 instead of 0x8005000 (due to reduce size of the bootloader is self),
-Note. upload to 0x8005000 was retained for backwards compatibility.
+2. In config.h there are per-board definitions for:
+* `LED_BANK`: GPIO Port (e.g. `GPIOA`, `GPIOB`, etc.)
+* `LED_PIN`: pin in that port (e.g. `1`)
+* `LED_ON_STATE`: whether the LED is lit when the pin is high (1) or low (0). This is because some generic boards have the LED between `Vcc` and the pin, hence are on when the `LED_PIN` is LOW.
 
+3. Boards with Maple hardware have an extra button (in addition to "Reset") and a special USB reset/disconnect hardware circuit. Such boards need a few additional defines, namely:
+** `BUTTON_BANK` and `BUTTON_PIN` for the extra button;
+** `USB_DISC_BANK` and `USB_DISC_PIN` for the USB disconnect circuit.
+For generic boards (ones that don't have `#define HAS_MAPLE_HARDWARE`), there is no need to define all this.
 
-If you have questions about the bootloader please raise and issue and I will attempt to answer them.
+## Other improvements on the original Maple bootloader
 
+1. Smaller footprint, now only 8k (this became possible due to code changes by [@jonatanolofsson](https://github.com/jonatanolofsson) which allowed the GCC optimisation for size flag to be used.
 
+2. Additional DFU AltID upload type was added, which allows the sketch to be loaded at 0x8002000 instead of 0x8005000 (due to reduced size of the bootloader itself). **Note** upload to 0x8005000 was retained for backwards compatibility.
 
+## Feedback
 
+If you have questions about the bootloader, please raise an issue and I will attempt to answer it.
```

Comment:
for questions the forum? http://www.stm32duino.com
and bugs a new issue?

Reply:
I would leave it as is (I am not changing this text in my commit, just moving it around), as it's author's prerogative to specify the ways to contact him.